integration/ig/k8s/top_block_io_test.go

```diff
@@ -16,33 +16,40 @@ package main
 
 import (
 	"fmt"
+	"strings"
 	"testing"
 
 	. "github.com/inspektor-gadget/inspektor-gadget/integration"
 	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/top/block-io/types"
-	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
 )
 
 func newTopBlockIOCmd(ns string, cmd string, startAndStop bool) *Command {
 	expectedOutputFn := func(output string) error {
 		expectedEntry := &types.Stats{
-			CommonData: eventtypes.CommonData{
-				Namespace: ns,
-				Pod:       "test-pod",
-			},
-			Comm:  "dd",
-			Write: true,
+			CommonData: BuildCommonData(ns, WithRuntimeMetadata(*containerRuntime)),
+			Comm:       "dd",
+			Write:      true,
 		}
 
 		normalize := func(e *types.Stats) {
-			e.Container = ""
+			// Docker and CRI-O uses a custom container name composed, among
+			// other things, by the pod UID. We don't know the pod UID in
+			// advance, so we can't match the exact expected container name.
+			prefixContainerName := "k8s_" + "test-pod" + "_" + "test-pod" + "_" + ns + "_"
+			if (*containerRuntime == ContainerRuntimeDocker || *containerRuntime == ContainerRuntimeCRIO) &&
+				strings.HasPrefix(e.Runtime.ContainerName, prefixContainerName) {
```

Comment:
That's the standard Docker and CRI-O uses in k8s. Let's see if there is any case it is not like that and we can add consider it here. Right now, the test will fail in case it doesn't use such a prefix so we will immediately realize about it.